extension.js

```diff
@@ -51,9 +59,36 @@ function maybe_enable_after_name_available() {
         statusNotifierWatcher = new StatusNotifierWatcher.StatusNotifierWatcher(watchDog);
 }
 
+function inner_enable() {
+    if (startupComplete && displayAvailable) {
+        isEnabled = true;
+        maybe_enable_after_name_available();
+    }
+}
+
 function enable() {
-    isEnabled = true;
-    maybe_enable_after_name_available();
+    // If the desktop is still starting up, we must wait until it is ready
+    if (Main.layoutManager._startingUp) {
+        startupPreparedId = Main.layoutManager.connect('startup-complete', () => {
+            Main.layoutManager.disconnect(startupPreparedId);
+            startupComplete = true;
+            inner_enable();
+        });
+    } else {
+        startupComplete = true;
+    }
+
+    // Ensure that the default Gdk Screen is available
+    if (Gtk.IconTheme.get_default() == null) {
```

Comment:
Maybe here would be better to check `!Gdk.DisplayManager.get().get_default_display()` instead?

Reply:
Gtk.IconTheme.get_default() fails because internally it can't get the default display. In some comments in #mutter they commented that Gtk.IconTheme should not rely on a display (or an screen... now I doubt), because it is an X11 specific structure. So, if in the future, they remove that, Gtk.IconTheme.get_default() will work directly and the extension won't have to wait.